instrumentation/host/host.go

```diff
@@ -0,0 +1,279 @@
+	c := Config{
+		MeterProvider: global.MeterProvider(),
+	}
+	for _, opt := range opts {
+		opt.ApplyHost(&c)
+	}
+	return c
+}
+
+// Start initializes reporting of host metrics using the supplied Config.
+func Start(c Config) error {
+	if c.MeterProvider == nil {
+		c.MeterProvider = global.MeterProvider()
+	}
+	h := &host{
+		meter: c.MeterProvider.Meter(
+			"host",
+			metric.WithInstrumentationVersion(contrib.SemVersion()),
+		),
+		config: c,
+	}
+	return h.register()
+}
+
+func (h *host) register() error {
+	var (
+		err error
+
+		processCPUTime metric.Float64SumObserver
+		hostCPUTime    metric.Float64SumObserver
+
+		hostMemoryUsage       metric.Int64UpDownSumObserver
+		hostMemoryUtilization metric.Float64UpDownSumObserver
+
+		networkIOUsage metric.Int64SumObserver
+
+		// lock prevents a race between batch observer and instrument registration.
+		lock sync.Mutex
+	)
+
+	proc, err := process.NewProcess(int32(os.Getpid()))
+	if err != nil {
+		return fmt.Errorf("could not find this process: %w", err)
+	}
+
+	lock.Lock()
+	defer lock.Unlock()
+
+	batchObserver := h.meter.NewBatchObserver(func(ctx context.Context, result metric.BatchObserverResult) {
+		lock.Lock()
+		defer lock.Unlock()
+
+		// This follows the OpenTelemetry Collector's "hostmetrics"
+		// receiver/hostmetricsreceiver/internal/scraper/processscraper
+		// measures User and System IOwait time.
+		// TODO: the Collector has per-OS compilation modules to support
+		// specific metrics that are not universal.
+		processTimes, err := proc.TimesWithContext(ctx)
+		if err != nil {
+			global.Handler().Handle(err)
+			return
+		}
+
+		hostTimeSlice, err := cpu.TimesWithContext(ctx, false)
+		if err != nil {
+			global.Handler().Handle(err)
+			return
+		}
+		if len(hostTimeSlice) != 1 {
+			global.Handler().Handle(fmt.Errorf("host CPU usage: incorrect summary count"))
+			return
+		}
+
+		vmStats, err := mem.VirtualMemoryWithContext(ctx)
+		if err != nil {
+			global.Handler().Handle(err)
+			return
+		}
+
+		ioStats, err := net.IOCountersWithContext(ctx, false)
+		if err != nil {
+			global.Handler().Handle(err)
+			return
+		}
+		if len(ioStats) != 1 {
+			global.Handler().Handle(fmt.Errorf("host network usage: incorrect summary count"))
+			return
+		}
+
+		// Process CPU time
+		result.Observe(LabelCPUTimeUser, processCPUTime.Observation(processTimes.User))
+		result.Observe(LabelCPUTimeSystem, processCPUTime.Observation(processTimes.System))
+
+		// Host CPU time
+		hostTime := hostTimeSlice[0]
+		result.Observe(LabelCPUTimeUser, hostCPUTime.Observation(hostTime.User))
+		result.Observe(LabelCPUTimeSystem, hostCPUTime.Observation(hostTime.System))
+
+		// TODO: "other" is a placeholder for actually dealing
+		// with these states.  Do users actually want this
+		// (unconditionally)?  How should we handle "iowait"
+		// if not all systems expose it?  Should we break
+		// these down by CPU?  If so, are users could to want
+		// to aggregate in-process?
```

Comment:
I think this is the correct initial approach :+1: 

If users want to have these exposed we can possibly add this to the `Config` as a something they can specify in the future.

```suggestion
		// TODO: "other" is a placeholder for actually dealing
		// with these states.  Do users actually want this
		// (unconditionally)?  How should we handle "iowait"
		// if not all systems expose it?  Should we break
		// these down by CPU?  If so, are users going to want
		// to aggregate in-process?
```